src/Sylius/Bundle/ResourceBundle/DependencyInjection/SyliusResourceExtension.php

```diff
@@ -31,84 +31,43 @@ class SyliusResourceExtension extends AbstractExtension
     public function load(array $config, ContainerBuilder $container)
     {
         $processor = new Processor();
-        $config    = $processor->processConfiguration(new Configuration(), $config);
+        $config = $processor->processConfiguration(new Configuration(), $config);
 
-        $this->loadServiceDefinitions($container, array(
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $configFiles = array(
             'services.xml',
             'storage.xml',
             'routing.xml',
-            'twig.xml',
-        ));
-
-        $classes = isset($config['resources']) ? $config['resources'] : array();
-
-        $container->setParameter('sylius.resource.settings', $config['settings']);
-
-        $this->createResourceServices($classes, $container);
+            'twig.xml'
+        );
 
-        $configClasses = array('default' => $this->getClassesFromConfig($classes));
-
-        if ($container->hasParameter('sylius.config.classes')) {
-            $configClasses = array_merge_recursive(
-                $configClasses,
-                $container->getParameter('sylius.config.classes')
-            );
+        foreach ($configFiles as $configFile) {
+            $loader->load($configFile);
         }
 
-        $container->setParameter('sylius.config.classes', $configClasses);
-    }
+        foreach ($config['resources'] as $alias => $resourceConfig) {
+            $metadata = Metadata::fromAliasAndConfiguration($alias, $resourceConfig);
 
-    /**
-     * @param array            $configs
-     * @param ContainerBuilder $container
-     */
-    private function createResourceServices(array $configs, ContainerBuilder $container)
-    {
-        $translationsEnabled = class_exists('Sylius\Bundle\TranslationBundle\DependencyInjection\Mapper');
+            $resources = $container->hasParameter('sylius.resources') ? $container->getParameter('sylius.resources') : array();
+            $resources = array_merge($resources, array($alias => $resourceConfig));
+            $container->setParameter('sylius.resources', $resources);
 
-        if ($translationsEnabled) {
-            $mapper = new Mapper();
-        }
+            DriverProvider::get($metadata)->load($container, $metadata);
 
-        foreach ($configs as $name => $config) {
-            list($prefix, $resourceName) = explode('.', $name);
-            $manager = isset($config['object_manager']) ? $config['object_manager'] : 'default';
+            if (isset($resourceConfig['translation'])) {
```

Comment:
there is an error here. 
you are no longer checking if the translation bundle exists.